[Filebeat] Add Zeek NTP Fileset

legoguy1000 opened the pull request:
## What does this PR do?

Add the NTP fileset to the Zeek Module

## Why is it important?

One of the remaining Zeek log files that isn't parsed already by Filebeat.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

elasticmachine commented:
## :grey_exclamation: Build Aborted
  
  > The PR is not allowed to run in the CI yet
  


<!-- BUILD BADGES-->
> _the below badges are clickable and redirect to their specific view in the CI or DOCS_
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-24224/detail/PR-24224/1//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-24224/detail/PR-24224/1//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-24224/detail/PR-24224/1//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-24224/detail/PR-24224/1//artifacts) 

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats


* Build Cause: `Pull request #24224 opened`


* Reason: `The PR is not allowed to run in the CI yet`


* Start Time: 2021-02-25T02:27:20.553+0000


* Duration: 4 min 37 sec


* Commit: b4688532c40cd55a1891ed94c291fe099eda55fa




#### Trends :test_tube:


![Image of Build Times](https://beats-ci.elastic.co/job/Beats/job/beats/job/PR-24224/buildTimeGraph/png)



</p>
</details>

<!-- TEST RESULTS IF ANY-->


<!-- STEPS ERRORS IF ANY -->









### Steps errors [![2](https://img.shields.io/badge/2%20-red)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-24224/detail/PR-24224/1//pipeline)
  <details><summary>Expand to view the steps failures</summary>
  <p>


  
  
  
  
##### `Load a resource file from a shared library`
<ul>
<li>Took 0 min 0 sec . View more details on <a href="https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/PR-24224/runs/1/steps/81/log/?start=0">here</a></li>
<li>Description: <code>approval-list/elastic/beats.yml</code></l1>
</ul>
  
  
  
  
##### `Error signal`
<ul>
<li>Took 0 min 0 sec . View more details on <a href="https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/PR-24224/runs/1/steps/84/log/?start=0">here</a></li>
<li>Description: <code>githubPrCheckApproved: The PR is not allowed to run in the CI yet. (Only users with write permissions can do so.)</code></l1>
</ul>
  
  </p>
  </details>



<!-- LOG OUTPUT-->
### Log output

<details><summary>Expand to view the last 100 lines of log output</summary>
<p>

```
[2021-02-25T02:28:58.512Z]  > git checkout -f b4688532c40cd55a1891ed94c291fe099eda55fa # timeout=15
[2021-02-25T02:29:00.163Z]  > git remote # timeout=10
[2021-02-25T02:29:00.167Z]  > git config --get remote.origin.url # timeout=10
[2021-02-25T02:29:00.171Z] using GIT_SSH to set credentials GitHub user @elasticmachine SSH key
[2021-02-25T02:29:00.174Z]  > git merge c5821d99bccea54f8253f3df247db3be4658834d # timeout=10
[2021-02-25T02:29:00.185Z]  > git rev-parse HEAD^{commit} # timeout=10
[2021-02-25T02:29:00.194Z]  > git config core.sparsecheckout # timeout=10
[2021-02-25T02:29:00.198Z]  > git checkout -f b4688532c40cd55a1891ed94c291fe099eda55fa # timeout=15
[2021-02-25T02:29:03.970Z] Commit message: "Add Zeek NTP Fileset"
[2021-02-25T02:29:03.970Z] First time build. Skipping changelog.
[2021-02-25T02:29:03.970Z] Cleaning workspace
[2021-02-25T02:29:04.302Z]  > git --version # timeout=10
[2021-02-25T02:29:04.305Z]  > git --version # 'git version 2.17.1'
[2021-02-25T02:29:04.309Z] fatal: bad object 8f30f60989444d11e43197dd15ee83662a4926cd
[2021-02-25T02:29:03.973Z]  > git rev-parse --verify HEAD # timeout=10
[2021-02-25T02:29:03.977Z] Resetting working tree
[2021-02-25T02:29:03.977Z]  > git reset --hard # timeout=10
[2021-02-25T02:29:04.217Z]  > git clean -fdx # timeout=10
[2021-02-25T02:29:05.035Z] Timeout set to expire in 3 hr 0 min
[2021-02-25T02:29:05.063Z] The timestamps step is unnecessary when timestamps are enabled for all Pipeline builds.
[2021-02-25T02:29:05.315Z] [INFO] Number of builds to be searched 10
[2021-02-25T02:29:06.057Z] [INFO] 'shallow' is forced to be disabled when running on PullRequests
[2021-02-25T02:29:06.082Z] Running in /var/lib/jenkins/workspace/Beats_beats_PR-24224/src/github.com/elastic/beats
[2021-02-25T02:29:06.119Z] [INFO] gitCheckout: Checkout SCM PR-24224 with default customisation from the Item.
[2021-02-25T02:29:06.151Z] [INFO] Override default checkout
[2021-02-25T02:29:06.212Z] Sleeping for 10 sec
[2021-02-25T02:29:16.246Z] The recommended git tool is: git
[2021-02-25T02:29:16.297Z] using credential f6c7695a-671e-4f4f-a331-acdce44ff9ba
[2021-02-25T02:29:16.328Z] Wiping out workspace first.
[2021-02-25T02:29:16.338Z] Cloning the remote Git repository
[2021-02-25T02:29:16.338Z] Using shallow clone with depth 10
[2021-02-25T02:29:16.338Z] Avoid fetching tags
[2021-02-25T02:29:16.355Z] Cloning repository git@github.com:elastic/beats.git
[2021-02-25T02:29:16.382Z]  > git init /var/lib/jenkins/workspace/Beats_beats_PR-24224/src/github.com/elastic/beats # timeout=10
[2021-02-25T02:29:16.388Z] Fetching upstream changes from git@github.com:elastic/beats.git
[2021-02-25T02:29:16.388Z]  > git --version # timeout=10
[2021-02-25T02:29:16.394Z]  > git --version # 'git version 2.17.1'
[2021-02-25T02:29:16.394Z] using GIT_SSH to set credentials GitHub user @elasticmachine SSH key
[2021-02-25T02:29:16.399Z]  > git fetch --no-tags --progress -- git@github.com:elastic/beats.git +refs/heads/*:refs/remotes/origin/* # timeout=15
[2021-02-25T02:29:36.317Z] Cleaning workspace
[2021-02-25T02:29:36.332Z] Using shallow fetch with depth 10
[2021-02-25T02:29:36.332Z] Pruning obsolete local branches
[2021-02-25T02:29:37.389Z] Merging remotes/origin/master commit c5821d99bccea54f8253f3df247db3be4658834d into PR head commit b4688532c40cd55a1891ed94c291fe099eda55fa
[2021-02-25T02:29:36.300Z]  > git config remote.origin.url git@github.com:elastic/beats.git # timeout=10
[2021-02-25T02:29:36.304Z]  > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
[2021-02-25T02:29:36.313Z]  > git config remote.origin.url git@github.com:elastic/beats.git # timeout=10
[2021-02-25T02:29:36.320Z]  > git rev-parse --verify HEAD # timeout=10
[2021-02-25T02:29:36.323Z] No valid HEAD. Skipping the resetting
[2021-02-25T02:29:36.323Z]  > git clean -fdx # timeout=10
[2021-02-25T02:29:36.335Z] Fetching upstream changes from git@github.com:elastic/beats.git
[2021-02-25T02:29:36.335Z] using GIT_SSH to set credentials GitHub user @elasticmachine SSH key
[2021-02-25T02:29:36.339Z]  > git fetch --no-tags --progress --prune -- git@github.com:elastic/beats.git +refs/pull/24224/head:refs/remotes/origin/PR-24224 +refs/heads/master:refs/remotes/origin/master # timeout=15
[2021-02-25T02:29:37.392Z]  > git config core.sparsecheckout # timeout=10
[2021-02-25T02:29:37.395Z]  > git checkout -f b4688532c40cd55a1891ed94c291fe099eda55fa # timeout=15
[2021-02-25T02:29:38.954Z] Merge succeeded, producing b4688532c40cd55a1891ed94c291fe099eda55fa
[2021-02-25T02:29:38.954Z] Checking out Revision b4688532c40cd55a1891ed94c291fe099eda55fa (PR-24224)
[2021-02-25T02:29:38.926Z]  > git remote # timeout=10
[2021-02-25T02:29:38.930Z]  > git config --get remote.origin.url # timeout=10
[2021-02-25T02:29:38.934Z] using GIT_SSH to set credentials GitHub user @elasticmachine SSH key
[2021-02-25T02:29:38.938Z]  > git merge c5821d99bccea54f8253f3df247db3be4658834d # timeout=10
[2021-02-25T02:29:38.949Z]  > git rev-parse HEAD^{commit} # timeout=10
[2021-02-25T02:29:38.957Z]  > git config core.sparsecheckout # timeout=10
[2021-02-25T02:29:38.962Z]  > git checkout -f b4688532c40cd55a1891ed94c291fe099eda55fa # timeout=15
[2021-02-25T02:29:40.398Z] Commit message: "Add Zeek NTP Fileset"
[2021-02-25T02:29:40.398Z] Cleaning workspace
[2021-02-25T02:29:40.573Z]  > git --version # timeout=10
[2021-02-25T02:29:40.577Z]  > git --version # 'git version 2.17.1'
[2021-02-25T02:29:40.581Z] fatal: bad object 8f30f60989444d11e43197dd15ee83662a4926cd
[2021-02-25T02:29:41.705Z] Masking supported pattern matches of $GIT_USERNAME or $GIT_PASSWORD
[2021-02-25T02:29:40.401Z]  > git rev-parse --verify HEAD # timeout=10
[2021-02-25T02:29:40.404Z] Resetting working tree
[2021-02-25T02:29:40.404Z]  > git reset --hard # timeout=10
[2021-02-25T02:29:40.492Z]  > git clean -fdx # timeout=10
[2021-02-25T02:29:42.394Z] + git fetch https://****:****@github.com/elastic/beats.git +refs/pull/*/head:refs/remotes/origin/pr/*
[2021-02-25T02:30:50.223Z] Running in /var/lib/jenkins/workspace/Beats_beats_PR-24224/src/github.com/elastic/beats/.git
[2021-02-25T02:30:50.423Z] Archiving artifacts
[2021-02-25T02:30:51.164Z] + git rev-parse HEAD
[2021-02-25T02:30:51.506Z] + git rev-parse HEAD
[2021-02-25T02:30:51.820Z] + git rev-parse origin/pr/24224
[2021-02-25T02:30:51.882Z] [INFO] githubEnv: Found Git Build Cause: pr
[2021-02-25T02:30:52.232Z] Masking supported pattern matches of $GITHUB_TOKEN
[2021-02-25T02:30:53.592Z] [WARN] githubApiCall: The REST API call https://api.github.com/repos/elastic/beats/pulls/24224/reviews return 0 elements
[2021-02-25T02:30:53.656Z] [INFO] githubPrCheckApproved: Title: Add Zeek NTP Fileset - User: legoguy1000 - Author Association: CONTRIBUTOR
[2021-02-25T02:30:53.990Z] ERROR: githubPrCheckApproved: The PR is not allowed to run in the CI yet
[2021-02-25T02:30:53.990Z] ERROR: githubPrCheckApproved: The PR is not allowed to run in the CI yet. (Only users with write permissions can do so.)
[2021-02-25T02:30:54.072Z] [INFO] Let's stop build #1. The PR is not allowed to run in the CI yet
[2021-02-25T02:30:54.094Z] Sleeping for 5 sec
[2021-02-25T02:30:55.091Z] Stage "Lint" skipped due to earlier failure(s)
[2021-02-25T02:30:55.169Z] Stage "Build&Test" skipped due to earlier failure(s)
[2021-02-25T02:30:55.249Z] Stage "Packaging" skipped due to earlier failure(s)
[2021-02-25T02:30:55.357Z] Running in /var/lib/jenkins/workspace/Beats_beats_PR-24224/src/github.com/elastic/beats
[2021-02-25T02:30:55.935Z] Running on Jenkins in /var/lib/jenkins/workspace/Beats_beats_PR-24224
[2021-02-25T02:30:56.203Z] [INFO] getVaultSecret: Getting secrets
[2021-02-25T02:30:56.286Z] Masking supported pattern matches of $VAULT_ADDR or $VAULT_ROLE_ID or $VAULT_SECRET_ID
[2021-02-25T02:30:57.440Z] + chmod 755 generate-build-data.sh
[2021-02-25T02:30:57.440Z] + ./generate-build-data.sh https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats/PR-24224/ https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats/PR-24224/runs/1 ABORTED 216615
[2021-02-25T02:30:57.691Z] INFO: curl https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats/PR-24224/runs/1/steps/?limit=10000 -o steps-info.json
[2021-02-25T02:30:58.242Z] INFO: curl https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats/PR-24224/runs/1/tests/?status=FAILED -o tests-errors.json
[2021-02-25T02:30:58.242Z] Retry 1/3 exited 22, retrying in 1 seconds...
[2021-02-25T02:30:59.153Z] Retry 2/3 exited 22, retrying in 2 seconds...

```

</p>
</details>



<!--COMMENT_GENERATED_WITH_ID_comment.id-->

legoguy1000 commented:
I'm trying to generate the expected log files but getting a weird timeout.  Have anyone seen this before??
```
>> python test: Integration Testing
Found Elastic Beats dir at /go/src/github.com/elastic/beats
exec: /go/src/github.com/elastic/beats/build/ve/docker/bin/pytest --timeout=90 --durations=20 -v --junit-xml=build/TEST-python-integration.xml tests/system/test_filebeat_xpack.py tests/system/test_http_endpoint.py tests/system/test_xpack_modules.py
======================================================================== test session starts =========================================================================
platform linux -- Python 3.7.3, pytest-6.0.1, py-1.9.0, pluggy-0.13.1 -- /go/src/github.com/elastic/beats/build/ve/docker/bin/python3
cachedir: .pytest_cache
rootdir: /go/src/github.com/elastic/beats, configfile: pytest.ini
plugins: rerunfailures-9.0, timeout-1.3.4
timeout: 90.0s
timeout method: signal
timeout func_only: True
collected 15 items / 14 deselected / 1 selected                                                                                                                      

tests/system/test_xpack_modules.py::XPackTest::test_fileset_file_0_zeek FAILED                                                                                 [100%]

============================================================================== FAILURES ==============================================================================
_________________________________________________________________ XPackTest.test_fileset_file_0_zeek _________________________________________________________________

a = (<test_xpack_modules.XPackTest testMethod=test_fileset_file_0_zeek>,)

    @wraps(func)
    def standalone_func(*a):
>       return func(*(a + p.args), **p.kwargs)

../../build/ve/docker/lib/python3.7/site-packages/parameterized/parameterized.py:518: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../filebeat/tests/system/test_modules.py:99: in test_fileset_file
    cfgfile=cfgfile)
../../filebeat/tests/system/test_modules.py:150: in run_on_file
    bufsize=0).wait()
/usr/lib/python3.7/subprocess.py:990: in wait
    return self._wait(timeout=timeout)
/usr/lib/python3.7/subprocess.py:1624: in _wait
    (pid, sts) = self._try_wait(0)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <subprocess.Popen object at 0x7f27237e7dd8>, wait_flags = 0

    def _try_wait(self, wait_flags):
        """All callers to this function MUST hold self._waitpid_lock."""
        try:
>           (pid, sts) = os.waitpid(self.pid, wait_flags)
E           Failed: Timeout >90.0s

/usr/lib/python3.7/subprocess.py:1582: Failed
------------------------------------------------------------------------ Captured stdout call ------------------------------------------------------------------------
Using elasticsearch: http://elasticsearch:9200
Testing zeek/ntp on /go/src/github.com/elastic/beats/x-pack/filebeat/module/zeek/ntp/test/ntp-json.log
------------------------------- generated xml file: /go/src/github.com/elastic/beats/x-pack/filebeat/build/TEST-python-integration.xml -------------------------------
======================================================================== slowest 20 durations ========================================================================
90.00s call     x-pack/filebeat/tests/system/test_xpack_modules.py::XPackTest::test_fileset_file_0_zeek

(2 durations < 0.005s hidden.  Use -vv to show these durations.)
====================================================================== short test summary info =======================================================================
FAILED tests/system/test_xpack_modules.py::XPackTest::test_fileset_file_0_zeek - Failed: Timeout >90.0s
============================================================ 1 failed, 14 deselected in 117.97s (0:01:57) ============================================================
>> python test: Integration Testing Complete
>>> Fixing file ownership issues from Docker at path=.
chown took: 34.8968867s, changed 0 files
>>> Fixing file ownership issues from Docker at path=/go/src/github.com/elastic/beats/build
chown took: 2m19.3409614s, changed 0 files
Error: running "/go/src/github.com/elastic/beats/build/ve/docker/bin/pytest --timeout=90 --durations=20 -v --junit-xml=build/TEST-python-integration.xml tests/system/test_filebeat_xpack.py tests/system/test_http_endpoint.py tests/system/test_xpack_modules.py" failed with exit code 1
ERROR: 1
exec: docker-compose -p filebeat_8_0_0_5356f8245a-snapshot rm --stop --force
Stopping filebeat_8_0_0_5356f8245a-snapshot_elasticsearch_1 ... done
Removing filebeat_8_0_0_5356f8245a-snapshot_beat_run_11032c46723d ... done
Removing filebeat_8_0_0_5356f8245a-snapshot_proxy_dep_1           ... done
Removing filebeat_8_0_0_5356f8245a-snapshot_elasticsearch_1       ... done
>> Done running testing inside of docker...
Teardown mage...
Error: running "docker-compose -p filebeat_8_0_0_5356f8245a-snapshot run -e DOCKER_COMPOSE_PROJECT_NAME=filebeat_8_0_0_5356f8245a-snapshot -e BEAT_STRICT_PERMS=false -e STACK_ENVIRONMENT=snapshot -e TESTING_ENVIRONMENT=snapshot -e GOCACHE=/go/src/github.com/elastic/beats/build/docker-gocache -v /home/alex/go/pkg/mod/cache/download:/gocache:ro -e GOPROXY=file:///gocache,direct -e EXEC_UID=1000 -e EXEC_GID=1000 -e GENERATE=true -e MAGEFILE_VERBOSE=1 -e BEATS_INSIDE_INTEGRATION_TEST_ENV=true -e GOFLAGS=-mod=readonly -e TESTING_FILEBEAT_FILESETS=ntp -e PYTEST_ADDOPTS=-k zeek beat /go/src/github.com/elastic/beats/x-pack/filebeat/build/mage-linux-amd64 pythonIntegTest" failed with exit code 1
```

legoguy1000 commented:
> I'm trying to generate the expected log files but getting a weird timeout. Have anyone seen this before??
> 
> ```
> >> python test: Integration Testing
> Found Elastic Beats dir at /go/src/github.com/elastic/beats
> exec: /go/src/github.com/elastic/beats/build/ve/docker/bin/pytest --timeout=90 --durations=20 -v --junit-xml=build/TEST-python-integration.xml tests/system/test_filebeat_xpack.py tests/system/test_http_endpoint.py tests/system/test_xpack_modules.py
> ======================================================================== test session starts =========================================================================
> platform linux -- Python 3.7.3, pytest-6.0.1, py-1.9.0, pluggy-0.13.1 -- /go/src/github.com/elastic/beats/build/ve/docker/bin/python3
> cachedir: .pytest_cache
> rootdir: /go/src/github.com/elastic/beats, configfile: pytest.ini
> plugins: rerunfailures-9.0, timeout-1.3.4
> timeout: 90.0s
> timeout method: signal
> timeout func_only: True
> collected 15 items / 14 deselected / 1 selected                                                                                                                      
> 
> tests/system/test_xpack_modules.py::XPackTest::test_fileset_file_0_zeek FAILED                                                                                 [100%]
> 
> ============================================================================== FAILURES ==============================================================================
> _________________________________________________________________ XPackTest.test_fileset_file_0_zeek _________________________________________________________________
> 
> a = (<test_xpack_modules.XPackTest testMethod=test_fileset_file_0_zeek>,)
> 
>     @wraps(func)
>     def standalone_func(*a):
> >       return func(*(a + p.args), **p.kwargs)
> 
> ../../build/ve/docker/lib/python3.7/site-packages/parameterized/parameterized.py:518: 
> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
> ../../filebeat/tests/system/test_modules.py:99: in test_fileset_file
>     cfgfile=cfgfile)
> ../../filebeat/tests/system/test_modules.py:150: in run_on_file
>     bufsize=0).wait()
> /usr/lib/python3.7/subprocess.py:990: in wait
>     return self._wait(timeout=timeout)
> /usr/lib/python3.7/subprocess.py:1624: in _wait
>     (pid, sts) = self._try_wait(0)
> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
> 
> self = <subprocess.Popen object at 0x7f27237e7dd8>, wait_flags = 0
> 
>     def _try_wait(self, wait_flags):
>         """All callers to this function MUST hold self._waitpid_lock."""
>         try:
> >           (pid, sts) = os.waitpid(self.pid, wait_flags)
> E           Failed: Timeout >90.0s
> 
> /usr/lib/python3.7/subprocess.py:1582: Failed
> ------------------------------------------------------------------------ Captured stdout call ------------------------------------------------------------------------
> Using elasticsearch: http://elasticsearch:9200
> Testing zeek/ntp on /go/src/github.com/elastic/beats/x-pack/filebeat/module/zeek/ntp/test/ntp-json.log
> ------------------------------- generated xml file: /go/src/github.com/elastic/beats/x-pack/filebeat/build/TEST-python-integration.xml -------------------------------
> ======================================================================== slowest 20 durations ========================================================================
> 90.00s call     x-pack/filebeat/tests/system/test_xpack_modules.py::XPackTest::test_fileset_file_0_zeek
> 
> (2 durations < 0.005s hidden.  Use -vv to show these durations.)
> ====================================================================== short test summary info =======================================================================
> FAILED tests/system/test_xpack_modules.py::XPackTest::test_fileset_file_0_zeek - Failed: Timeout >90.0s
> ============================================================ 1 failed, 14 deselected in 117.97s (0:01:57) ============================================================
> >> python test: Integration Testing Complete
> >>> Fixing file ownership issues from Docker at path=.
> chown took: 34.8968867s, changed 0 files
> >>> Fixing file ownership issues from Docker at path=/go/src/github.com/elastic/beats/build
> chown took: 2m19.3409614s, changed 0 files
> Error: running "/go/src/github.com/elastic/beats/build/ve/docker/bin/pytest --timeout=90 --durations=20 -v --junit-xml=build/TEST-python-integration.xml tests/system/test_filebeat_xpack.py tests/system/test_http_endpoint.py tests/system/test_xpack_modules.py" failed with exit code 1
> ERROR: 1
> exec: docker-compose -p filebeat_8_0_0_5356f8245a-snapshot rm --stop --force
> Stopping filebeat_8_0_0_5356f8245a-snapshot_elasticsearch_1 ... done
> Removing filebeat_8_0_0_5356f8245a-snapshot_beat_run_11032c46723d ... done
> Removing filebeat_8_0_0_5356f8245a-snapshot_proxy_dep_1           ... done
> Removing filebeat_8_0_0_5356f8245a-snapshot_elasticsearch_1       ... done
> >> Done running testing inside of docker...
> Teardown mage...
> Error: running "docker-compose -p filebeat_8_0_0_5356f8245a-snapshot run -e DOCKER_COMPOSE_PROJECT_NAME=filebeat_8_0_0_5356f8245a-snapshot -e BEAT_STRICT_PERMS=false -e STACK_ENVIRONMENT=snapshot -e TESTING_ENVIRONMENT=snapshot -e GOCACHE=/go/src/github.com/elastic/beats/build/docker-gocache -v /home/alex/go/pkg/mod/cache/download:/gocache:ro -e GOPROXY=file:///gocache,direct -e EXEC_UID=1000 -e EXEC_GID=1000 -e GENERATE=true -e MAGEFILE_VERBOSE=1 -e BEATS_INSIDE_INTEGRATION_TEST_ENV=true -e GOFLAGS=-mod=readonly -e TESTING_FILEBEAT_FILESETS=ntp -e PYTEST_ADDOPTS=-k zeek beat /go/src/github.com/elastic/beats/x-pack/filebeat/build/mage-linux-amd64 pythonIntegTest" failed with exit code 1
> ```

I'm running `GENERATE=true PYTEST_ADDOPTS="-k zeek" TESTING_FILEBEAT_FILESETS=ntp mage -v pythonIntegTest` from `beats/x-pack/filebeat`

elasticmachine commented:
Pinging @elastic/security-external-integrations (Team:Security-External Integrations)

leehinman commented:
Thanks this looks awesome.

The easiest way to generate the expected golden files is to let the mage target handle it.  In the `x-pack/filebeat` directory run:

```
GENERATE=true TESTING_FILEBEAT_MODULES=zeek TESTING_FILEBEAT_FILESETS=ntp mage -v pythonIntegTest
```

That will generate the golden files, limiting it to just the zeek module and the ntp fileset.

legoguy1000 commented:
> Thanks this looks awesome.
> 
> The easiest way to generate the expected golden files is to let the mage target handle it. In the `x-pack/filebeat` directory run:
> 
> ```
> GENERATE=true TESTING_FILEBEAT_MODULES=zeek TESTING_FILEBEAT_FILESETS=ntp mage -v pythonIntegTest
> ```
> 
> That will generate the golden files, limiting it to just the zeek module and the ntp fileset.

Tried this too, still getting a timeout error.  I think its just me trying to run this on a WSL 2 VM.  THough didn't have this issue with my previous PR.  Going to try on a dedicated system.

legoguy1000 commented:
Its still timing out as shown in the post above.  I can't seem to figure out what's causing it.

leehinman commented:
> Its still timing out as shown in the post above. I can't seem to figure out what's causing it.

turns out in manifest.yml you have:
```
ingest_pipeline: ingest/pipeline.json
```

it should be `ingest/pipeline.yml` after fixing that the mage target worked for me, although it did find a field that wasn't documented.

legoguy1000 commented:
> > Its still timing out as shown in the post above. I can't seem to figure out what's causing it.
> 
> turns out in manifest.yml you have:
> 
> ```
> ingest_pipeline: ingest/pipeline.json
> ```
> 
> it should be `ingest/pipeline.yml` after fixing that the mage target worked for me, although it did find a field that wasn't documented.

🤦‍♂️ 🤦‍♂️ I have updated the code and it should be ready to run the pipelines

legoguy1000 commented:
@leehinman  @andrewkroh  can 1 of u run the jenkins pipeline?

andrewkroh commented:
jenkins, run tests

andrewkroh commented on x-pack/filebeat/module/zeek/ntp/_meta/fields.yml:
```diff
@@ -0,0 +1,63 @@
+    - name: poll
+      type: double
+      description: >
+        The maximum interval between successive messages.
```
For `poll`, `precision`, `root_delay`, and `root_disp` can you please include a unit of measurement in the description.

andrewkroh replied:
Looking at the Zeek source those appear be in seconds always. https://github.com/zeek/zeek/blob/3819146511b173513d862cb1b45e4f9bc73548ec/src/analyzer/protocol/ntp/ntp-analyzer.pac#L24-L43

legoguy1000 replied:
I was just about to do the updates.  beat me be mere minutes :).  Appreciate the assist.

andrewkroh commented on CHANGELOG.next.asciidoc:
```diff
@@ -877,6 +877,8 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...master[Check the HEAD d
 - Upgrade okta to ecs 1.8.0 and move js processor to ingest pipeline {issue}23118[23118] {pull}23929[23929]
 - Update zoom module to ECS 1.8. {pull}23904[23904] {issue}23118[23118]
 - Support X-Forwarder-For in IIS logs. {pull}19142[192142]
+- Added NTP fileset to Zeek module {pull}23772[23772]
```
This one looks like it's extra from copy/paste issue.

andrewkroh reviewed (commented):
LGTM, and tests are passing. I just have minor comments to improve the docs.

andrewkroh reviewed (approved):
LGTM. I updated the docs.

andrewkroh commented:
jenkins, run tests
